packages/api/lib/executions.js

```diff
@@ -57,4 +89,126 @@ async function describeGranuleExecution(executionArn, stepFunctionUtils = StepFu
   return executionDescription;
 }
 
-module.exports = { chooseTargetExecution, describeGranuleExecution };
+/**
+ * Retrieves the Cumulus ID of a collection from the Postgres database.
+ *
+ * @param {Knex | KnexTransaction} knex - The Knex client object for interacting with the database.
+ * @param {string} apiCollection - The collection object from the API.
+ * @returns {Promise<number>} A promise that resolves to the cumulus_id of the collection.
+ * @throws {Error} Throws an error if the collection could not be found in the database.
+ */
+const _getCollectionCumulusId = async (knex, apiCollection) => {
+  const collectionId = deconstructCollectionId(apiCollection);
+  const collectionPgModel = new CollectionPgModel();
+
+  log.info(`Querying name: ${collectionId.name}, version: ${collectionId.version}`);
+  try {
+    const collectionCumulusId = await collectionPgModel.getRecordCumulusId(knex, {
+      name: collectionId.name,
+      version: collectionId.version,
+    });
+    if (!isNumber(collectionCumulusId)) {
+      throw new Error(`Internal Error: Collection ID ${collectionCumulusId} is not a number`);
+    }
+    log.info(`Collection ID: ${JSON.stringify(collectionCumulusId)}`);
+    return collectionCumulusId;
+  } catch (error) {
+    if (error instanceof RecordDoesNotExist) {
+      log.error(`Collection ${collectionId.name} version ${collectionId.version} not found in database`);
+    }
+    throw error;
+  }
+};
+
+/**
+ * Deletes execution records from the RDS database using
+ * batchDeleteExecutionFromDatabaseByCumulusCollectionId.
+ *
+ * @param {Object} params - The parameters object.
+ * @param {Object} params.knex - The Knex client object for interacting with the database.
+ * @param {number} params.collectionCumulusId - The ID of the collection whose execution records
+ *  are to be deleted.
+ * @param {number} params.batchSize - The number of records to delete in each batch.
+ * @returns {Promise<number>} A promise that resolves to the number of records deleted.
+ * @throws {Error} Throws an error if deletion fails.
+ */
+const _deleteRdsExecutionsFromDatabase = async ({
+  knex,
+  collectionCumulusId,
+  batchSize,
+}) => await pRetry(
+  async () => {
+    const batchDeleteResult = await batchDeleteExecutionFromDatabaseByCumulusCollectionId({
+      knex,
+      collectionCumulusId,
+      batchSize,
+    });
+    log.info(
+      `Deleted ${batchDeleteResult} execution records from RDS for collection ${collectionCumulusId}`
+    );
+    return batchDeleteResult;
+  },
+  {
+    retries: 3,
+    minTimeout: 60000,
+    maxTimeout: 90000,
+    onFailedAttempt: (error) => {
```

Comment:
> Dumb question but do we need retry logic here?

I think so - we don't want failures for things like scaling to kill the async operation outright.   

> it's possible you could swamp your db connection pool

Swamping the connection pool is unlikely - this is using a single query per batch in serial.

Reply:
> Is it worth retrying in specific error cases? Like when the knex pool is full or.... other?

Possibly - it feels like enumerating all of the possible retries is harder than *not* retrying if we have specific error cases we want to avoid.

Reply:
Fair enough. I'm not sure how to evaluate retries on a bulk operation like this. We've hit problems where we're retrying when we know it will fail but that's certainly better than not retrying when we think it'll be successful.